### PR TITLE
Create runtimebase image

### LINKED ISSUE #58: Separate base and build-base images
Comes from https://github.com/sociomantic-tsunami/dhtnode/pull/136#discussion_r202753499 - current `base` image is a bit too "fat" for deployment purpose as it install bunch of tools used for development. The real base should simply add sociomantic bintray repos and GPG key and stop at that.

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
This image can be used as a base image for running applications (as opposed to
build and test applications, as the current images).

It only installs a very small number of packages and apt repos on top of a base
`ubuntu` image.

In theory the development images could be based on this one, but this
one is kept separated to ease the bootstrapping of new Ubuntu versions,
as to create tools that need to be shipped by the `runtimebase` and
`develdlang` images we need to first have a base development image to
make those packages, so we need to break the circular dependency
somewhere.

Fixes #58.

### CODE REVIEW

File: relnotes/rename.migration.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+### Images renamed with a `devel` prefix
+
+To make the distinction between runtime and development images, the current
+images are renamed with the `devel` prefix:
+
+* `base` -> `develbase`
+* `dlang` -> `develdlang`
+
+Make sure to update the image name too in your `FROM` lines when upgrading the
+version number.
```
Comment (nemanja-boric-sociomantic) [inline relnotes/rename.migration.md:10]: It somehow sounds better if they would be `basedevel` bzw. `dlangdevel`.

File: docker/develbase [hunk 2]
```diff
@@ -8,49 +8,21 @@ set -xeu
 # Get utility stuff
 . ./util.sh
```
Comment (nemanja-boric-sociomantic) [inline docker/develbase:9]: I think you've missed to checking the changes inside `util.sh`?

Comment (nemanja-boric-sociomantic) [reply docker/develbase:9]: Also this commit has `base-apt-preferenced` rename checked in.

File: docker/util.sh [hunk 3]
```diff
@@ -55,6 +55,51 @@ gem_ver()
 	)
 }
 
+apt_update_and_install_base_packages()
```
Comment (nemanja-boric-sociomantic) [inline docker/util.sh:58]: This should be part of the previous commit.

File: develdlang.Dockerfile [hunk 4]
```diff
@@ -32,4 +32,4 @@ LABEL \
     com.sociomantic.version.hmod=$VERSION_HMOD
 
 COPY docker/ /docker-tmp
-RUN cd /docker-tmp && ./dlang && rm -fr /docker-tmp
+RUN cd /docker-tmp && ./develdlang && rm -fr /docker-tmp
```
Comment (mihails-strasuns) [inline develdlang.Dockerfile:35]: Will there ever be non-devel `dlang` image?

File: relnotes/rename.migration.md (hunk 1 above)
Comment (leandro-lucarella-sociomantic) [reply relnotes/rename.migration.md:10]: But then all devel images get sorted together in the FS and such, this is why I prefer prefixes. It feels more hierarchical (from more general to more specific).

File: docker/develbase (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [reply docker/develbase:9]: Dammit, I messed up with rebases and I might have lost them unless I can recover stashes...

File: develdlang.Dockerfile (hunk 4 above)
Comment (leandro-lucarella-sociomantic) [reply develdlang.Dockerfile:35]: I was going for consistency, to indicate it's based on the `develbase`, but apparently the new, consistent, names are a bit controversial.

File: docker/develbase (hunk 2 above)
Comment (nemanja-boric-sociomantic) [reply docker/develbase:9]: Don't worry I see it's in the next commit :-)

Comment (leandro-lucarella-sociomantic) [reply docker/develbase:9]: Ah, it was in another commit, I could recover it with `git fsck` anyway (/me hugs `git`)

**Comment by leandro-lucarella-sociomantic** [comment]
Green! Ping!

**Comment by nemanja-boric-sociomantic** [comment]
My only non-blocking concern is naming. I'm more in favor of `*-[devel,deploy]` but it's really not a blocker from my side if you like this order and it's fine for @mihails-strasuns-sociomantic

**Comment by mihails-strasuns** [comment]
My preferences are not strong enough to push either way :) Feel free to self-merge wit hwhatever naming you decide on.

**Review by mihails-strasuns — APPROVED** [review]

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Comment by leandro-lucarella-sociomantic** [comment]
About naming I don't find an improvement noticeable enough to justify the extra work to rename, if I would have came up with what nemanja suggests at first, I would be OK with it. I will just merge.

**Pull request merged**